lib/Slic3r/GUI/Plater/SplineControl.pm

```diff
@@ -267,7 +267,11 @@ sub _update_canvas_size {
     my @size = ($canvas_w - 2*$padding, $canvas_h - 2*$padding);
     $self->{canvas_size} = [@size];
     
-    $self->{scaling_factor_x} = $size[0]/($self->{max_layer_height} - $self->{min_layer_height});
+    if ($self->{max_layer_height} == $self->{min_layer_height}) {
```

Comment:
These are floats, aren't they? I think you shouldn't use an equality comparison here.

```suggestion
    if (abs($self->{max_layer_height} - $self->min_layer_height) < 0.000001) {
```

Reply:
You can use EPSILON instead of "0.000001"

Reply:
Did it with a comparator as @platsch suggested (which makes more sense here than my original test).